include/pagmo_plugins_nonfree/worhp.hpp

```diff
@@ -0,0 +1,1203 @@
+                  });
+
+        // NOTE: Worhp requires a single sparsity pattern for the hessian of the lagrangian (that is,
+        // the pattern must be valid for objfun and all constraints), but we provide a separate sparsity pattern for
+        // objfun and every constraint. We will thus need to merge our sparsity patterns in a single sparsity
+        // pattern.
+        sparsity_pattern merged_hs;
+        // Store the original hessians sparsity only if it is user-provided.
+        auto hs = prob.hessians_sparsity();
+        if (prob.has_hessians_sparsity()) {
+            for (const auto &sp : hs) {
+                // NOTE: we need to create a separate copy each time as std::set_union() requires distinct ranges.
+                const auto old_merged_hs(merged_hs);
+                merged_hs.clear();
+                std::set_union(old_merged_hs.begin(), old_merged_hs.end(), sp.begin(), sp.end(),
+                               std::back_inserter(merged_hs));
+            }
+        } else {
+            // If the hessians sparsity is not user-provided, dense patterns are assumed.
+            merged_hs = detail::dense_hessian(prob.get_nx());
+        }
+        // -------------------------------------------------------------------------------------------------------------------------
+        /*
+         * In WORHP the HM sparsity requires lower triangular entries first,
+         * then all the diagonal elements (also the zeros) (cani maledetti^2)
+         */
+        // Create the corresponding index map between pagmo and worhp sparse representation of the lower triangular part
+        // of the hessian of the lagrangian
+        std::vector<vector_double::size_type> hs_idx_map(merged_hs.size());
+        std::iota(hs_idx_map.begin(), hs_idx_map.end(), 0);
+        // Sort the resulting hessian of the lagrangian sparsity according to worhp twisted choice.
+        // Lexicographic from right to left, i.e. ((1,0),(2,0),(0,1), )
+        std::sort(hs_idx_map.begin(), hs_idx_map.end(),
+                  [&merged_hs](vector_double::size_type &idx1, vector_double::size_type &idx2) -> bool {
+                      return (merged_hs[idx1].second < merged_hs[idx2].second
+                              || (!(merged_hs[idx2].second < merged_hs[idx1].second)
+                                  && merged_hs[idx1].first < merged_hs[idx2].first));
+                  });
+
+        // We remove the diagonal entries from the hessian sparsity as merged from pagmo (if present)
+        auto it2 = std::remove_if(hs_idx_map.begin(), hs_idx_map.end(),
+                                  [&merged_hs](std::vector<vector_double::size_type>::size_type &idx) -> bool {
+                                      return (merged_hs[idx].first == merged_hs[idx].second);
+                                  });
+        hs_idx_map.resize(it2 - hs_idx_map.begin());
```

Comment:
I'd use the ``erase()`` method rather than resize.

Reply:
ok, sure.